release-21.1: colexecerror: catch panics from packages in sql/sem folder

yuzefovich opened the pull request:
Backport 1/1 commits from #62889.
Backport 1/1 commits from #63178.

/cc @cockroachdb/release

---

Previously, we would only catch panics from `sql/sem/tree` package.
Recently sqlsmith encountered a crash because of a panic in
`sql/sem/builtins` package, and I believe it is reasonable to catch
panics from that package as well as from `sql/sem/transform`, so we will
now be catching based on `sql/sem` prefix.

Addresses: #62846.

Release note: None

**sem/builtins: vectorized engine no longer catches crdb_internal.force_panic**

Previously, when executing `crdb_internal.force_panic` builtin, if it
was executed via the vectorized engine, we would catch the panic and
convert it into an internal error instead. This is undesirable and is
now fixed.

Release note: None

yuzefovich committed:
colexecerror: catch panics from packages in sql/sem folder

Previously, we would only catch panics from `sql/sem/tree` package.
Recently sqlsmith encountered a crash because of a panic in
`sql/sem/builtins` package, and I believe it is reasonable to catch
panics from that package as well as from `sql/sem/transform`, so we will
now be catching based on `sql/sem` prefix.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62913)
<!-- Reviewable:end -->

yuzefovich commented:
I believe it is worth it backporting the PR to 21.1, but it doesn't fix any GA-blockers, so I wonder whether we should backport it for 20.1.1.

yuzefovich commented:
Will wait for 21.1.1.

yuzefovich committed:
sem/builtins: vectorized engine no longer catches crdb_internal.force_panic

Previously, when executing `crdb_internal.force_panic` builtin, if it
was executed via the vectorized engine, we would catch the panic and
convert it into an internal error instead. This is undesirable and is
now fixed.

Release note: None

yuzefovich commented:
Friendly ping @jordanlewis
